src/libnyoci/nyoci-outbound.c

```diff
@@ -320,7 +320,7 @@ nyoci_outbound_add_options_up_to_key_(
 
 #if NYOCI_CONF_TRANS_ENABLE_OBSERVING
 	if ( (self->current_transaction != NULL)
-	  && (self->current_transaction->flags & NYOCI_TRANSACTION_OBSERVE == NYOCI_TRANSACTION_OBSERVE)
+	  && ((self->current_transaction->flags & NYOCI_TRANSACTION_OBSERVE) == NYOCI_TRANSACTION_OBSERVE)
```

Comment:
Good catch! Mind pulling this commit out into it's own pull request?

Reply:
I can't claim credit -- GCC found it and warned me (but I can't remember which warning flag it was.)
I'll submit a separate PR when I have a chance.